Add Modal metrics data to integration (#2552)
DataDog/integrations-extras: modal/README.md

Reviewer: On line 65, please add the following:
```
[5]: https://github.com/DataDog/integrations-extras/blob/master/modal/metadata.csv
```

Author: Nice thanks

### Metrics

> Modal publishes container metrics to Datadog.

Reviewer 2: ```suggestion
See [metadata.csv][3] for a list of metrics included with this integration.
```

Author: done

> See [metadata.csv][5] for a list of metrics included with this integration.

Reviewer 3: Not nitpicking your writing here - I think we need to use exact wording to ensure the "magic phrase" works properly:
```suggestion
See [metadata.csv][5] for a list of metrics provided by this integration.
```
